app/grandchallenge/challenges/tasks.py

```diff
@@ -55,12 +58,34 @@ def update_challenge_results_cache():
     )
 
 
+def send_alert_if_budget_consumed_warning_threshold_exceeded(challenge):
+    if (
+        challenge.has_changed("compute_cost_euro_millicents")
+        and challenge.approved_compute_costs_euro_millicents
+    ):
+        for threshold in sorted(
+            challenge.percent_budget_consumed_warning_thresholds, reverse=True
+        ):
+            if (
+                challenge.initial_value("compute_cost_euro_millicents")
+                < challenge.approved_compute_costs_euro_millicents
+                * threshold
+                / 100
+                <= challenge.compute_cost_euro_millicents
+            ):
```

Comment:
I find this if statement hard to grok. Can you maybe assign some variables to help with readability? Style wise, it is usually easiest for other developers to have everything laid out for them, and the if statement be quite straightforward:

```python
if should_do_something:
    do_something()
```

then `should_do_something` is built up through logic:

```python
it_is_sunday = datetime.datetime.now().weekday() == 6
feeling_under_the_weather = random.random() < 0.05

should_do_something = not it_is_sunday and not feeling_under_the_weather
```

Reply:
Just to confirm: I also find that hard to read. Black did not do this statement justice. All the operators "seem to be equal" even though this is an example of the rare ternary comparison... hard to parse!

Reply:
Yeah, sorry about this one. I was optimizing the code and it got a bit out of hand. And then, as Paul said, Black came in and made matters even worse... I'll clean it up!

Reply:
It should be a lot better now.